### ISSUE TITLE
Browser Age Calculation Issue?

### ISSUE THREAD

**Issue by ForbiddenEra** [browser-update/browser-update#553]
I attempted to update some current versions of browsers in the script, for example I set

```
this.vsakt={c:"102",f:"101",s:"15.4",e:"102",i:"12",ios:"15.0",samsung:"17.0",o:"88",e_a:"100",o_a:"69",y:"22.5.3",v:"5.2",uc:"13.4.0",brave:"1.39.122"};
```

to reflect the current available versions and added brave to the list as I noticed it seems to be detectable although I believe it's possibly ignored at the moment (although, I would like to also target that particular browser)

Anyway, when the detection comes back with those values, I get:
`age_years: 0` for Chrome; which is probably what one would expect for a current browser; however, if the script is not updated and is the same as is currently released:

```
this.vsakt={c:"98",f:"97",s:"15.2",e:"98",i:"12",ios:"15.0",samsung:"15.0.2",o:"83",e_a:"98",o_a:"66.2",y:"22.1.0",v:"5.1",uc:"13.4.0"};
```

then I get `age_years: -0.5` .. this is definitely not consistent; the browser I am running is not suddenly older simply because it's version is greater/higher than what `browser-update` considers current.

It seems that the code that does this is actually relying on what the script thinks the current version is; this partially harks back to my other issue #551 as a partial root cause with the browser metadata not being updated.

The code at https://github.com/browser-update/browser-update/blob/3ee0fecb52123635d6965207758e99d520cdcb22/scripts/update.npm.full.js#L150 says:

```
        r.age_years=Math.round(((s.vsakt[r.n]-r.v)/releases_per_year[r.n])*10)/10 || 0
```

However, there's nothing preventing `r.v` from being either greater or less than `s.vsakt[r.n]`; perhaps it should check if `r.v` is greater than `s.vsakt[r.n]` and return `0` in that case, as if the detected version is greater than the known current version then it's likely not that old and a safe value.

Thoughts?

**Comment by josselex** [comment]
The Problem with Brave Browser is that it does currently not properly identify itself.

**Comment by hortoncallum155-ops** [comment]
Good catch on this inconsistency, negative age values can definitely create confusion in version-based detection logic.

While working with date difference and age calculation logic in one of my web tools, I noticed how higher version values or updated inputs can sometimes result in unexpected outputs due to rounding and comparison conditions. Situations like leap year handling, version gap mapping and release cycle assumptions often affect the final age result if boundary checks are not properly defined.

In most cases, understanding [how](https://myagecalculate.com/) the final age calculation process reacts to dynamic input helps maintain accurate output during data processing and testing environments. This becomes especially important when working with real-time user input or automated detection modules.

Might be worth considering a condition to prevent calculation drift when detected values exceed expected baselines.
